### PR TITLE
*: make unified-pool use FuturePool

### PR DESCRIPTION
**Pull request by nolouch**
### What is changed and how it works?

In TiKV, we use a FuturePool abstraction to facilitate the addition of customized behavior, such as metrics. However, in some places like unified-pool, it is not utilized and instead yatp is used directly.

make unified-pool use `FuturePool`

Issue Number: Close https://github.com/tikv/tikv/issues/15924

What's Changed:

```commit-message
*: make unified-pool use FuturePool 
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/tikv_util/src/yatp_pool/mod.rs [hunk 1]
```diff
@@ -377,6 +377,17 @@ impl<T: PoolTicker> YatpPoolBuilder<T> {
...
+            .name_prefix
+            .clone()
+            .unwrap_or_else(|| "yatp_pool".to_string());
+        let size = self.core_thread_count;
+        let task = self.max_tasks;
+        let pool = self.build_multi_level_pool();
+        FuturePool::from_pool(pool, &name, size, task)
+    }
+
     pub fn build_multi_level_pool(self) -> ThreadPool<TaskCell> {
```
Comment (bufferflies) [inline components/tikv_util/src/yatp_pool/mod.rs:391]: where is it used after this pr, only test case?

Comment (nolouch) [reply components/tikv_util/src/yatp_pool/mod.rs:391]: `build_multi_level_future_pool` also use it.

**Review by bufferflies — APPROVED** [review]

File: src/read_pool.rs [hunk 2]
```diff
@@ -106,7 +104,7 @@ pub enum ReadPoolHandle {
         read_pool_low: FuturePool,
     },
     Yatp {
-        remote: Remote<TaskCell>,
+        remote: FuturePool,
```
Comment (glorv) [inline src/read_pool.rs:107]: FuturePool already implement running tasks statistics, scaling tasks, so you should remove the implement in yatp handle and use FuturePool's directly.

**Review by glorv — COMMENTED** [review]
rest LGTM.

**Comment by glorv** [comment]
Maybe we should deprecate the `FuturePools` implement and only support yatp in the near future. I don't see any user cases that still uses the `FuturePools`. @Connor1996  what do you think?

File: src/read_pool.rs (hunk 2 above)
Comment (nolouch) [reply src/read_pool.rs:107]: The metric `running_task` is still being saved for compatibility reasons.

Comment (glorv) [reply src/read_pool.rs:107]: There is a metric with the same function in FuturePoolInner.env called `metrics_running_task_count`, you can use it to replace the running tasks in Yatp handle.

**Comment by nolouch** [comment]
Before:
```
src/read_pool.rs[ReadPool] -> wrap(yatp)
src/future_pool.rs[FuturePool] -> wrap(yatp)
```

After:
```
src/read_pool.rs[ReadPool] -> wrap(FuturePool) -> wrap(yatp)
src/future_pool.rs[FuturePool] -> wrap(yatp)
```


I think maybe we can remove `src/read_pool` in the future. current, we unify the underlayer is `FuturePool` first. Additionally, `FuturePool` is utilized in other areas such as snapshot worker, ttl worker, and deadlock-checker worker. Therefore, I think that FuturePool has a wider range than direct Yatp or ReadPool in tikv.

File: src/read_pool.rs (hunk 2 above)
Comment (glorv) [reply src/read_pool.rs:107]: Ok. Then I think you can at least remove `max_tasks` and `pool_size` as they are used for scaling pool size. BTW, please add a deprecated comment for `running_tasks` so we may permanently remove it in v8.0.

**Review by glorv — APPROVED** [review]

**Pull request merged**